entry.tsx

```diff
@@ -32,10 +32,7 @@ function preRenderSetup(callwhendone: () => void) {
             return;
         }
 
-        let displayable = false;
-        if (isDevMode()) {
-            displayable = true;
-        }
+        const displayable = store.getState().entities.general.config.EnableDeveloper === 'true';
```

Comment:
this is the file where i dont think there is anyone way to access redux store rather than directly, since this is webpack hook which is run before pre-rendered application is initialized in module federation.

Reply:
do we have access to store too here, since this is pre rendering the root component

Reply:
Instead of looking at this here, what if we just checked `EnableDeveloper` in the `getDisplayableErrors` selector when the error has a `DEVELOPER` type? Alternatively, we could have `logError` do something similar and override the `displayable` argument passed in here

Reply:
yes having the displayable in the logError action seems like a good way to do it. I will make the change.

Reply:
So assuming we do this inside of logError action we have two options
- Remove the displayable paramater itself from the actions `logError(error: ServerError, displayable, consoleError)` and let it be handled by state's EnableDeveloper . But with this every logError would start displaying error in announcement bar if developer mode is on. Do we want that?
- Have the displayable paramter as it is but be optional. So when user passes it as true logical and it with the store's EnableDeveloper. If both are true then that what is passed in `getLogErrorAction`

Reply:
>     * Remove the displayable paramater itself from the actions `logError(error: ServerError, displayable, consoleError)` and let it be handled by state's EnableDeveloper . But with this every logError would start displaying error in announcement bar if developer mode is on. Do we want that?

That's sort of what I meant by my first suggestion. We only pass `displayable` as true from here and from the GraphQL actions you wrote. If we ignore the GraphQL stuff for the moment since that behaves differently from other API responses, this code is also the only place that we pass an object with `type: AnnouncementBarTypes.DEVELOPER` into `logError` as well, so we could just have the error bar component only show errors with that `type` if `EnableDeveloper` is true. Otherwise, it behaves as before where it doesn't show those errors.

Does that make sense? I'm kind of 0/5 on if we need to change this further since this should be an edge case anyway since it's part of the error handler

Reply:
there seems to be other places also where we are setting explicitly 
1.
https://github.com/mattermost/mattermost-webapp/blob/efb1211400634e465cb422270b3f99f84a389d8c/actions/websocket_actions.jsx#L308-L310

2.
https://github.com/mattermost/mattermost-webapp/blob/87d4d3ac9f620bb9be446c0442da526b609fef7c/actions/views/posts.js#L22-L33

Reply:
I moved the isDevMode to the action and with the check for developer error.